### ISSUE TITLE
Why is it not supported on mono to set the value of function pointer through reflection

### ISSUE THREAD

**Issue by Liangjia0411** [dotnet/runtime#107070]
This is my code:
```c#
public class TempTest
{
    public static void Run()
    {
        var type = typeof(TempTest);
        var field = type.GetField(nameof(TempTest.UnmangedFunctionPointer), BindingFlags.Static | BindingFlags.Public);
        field.SetValue(null, null);
    }

    public static unsafe delegate* unmanaged<void> UnmangedFunctionPointer;
}
```
When I call `TempTest.Run`, a crash will occur: `type 0x1b not handled in ves_icall_FieldInfo_SetValueInternal`.
I See the `type->type` in `ves_icall_FieldInfo_SetValueInternal` is `MONO_TYPE_FNPTR (27)`.

**Comment by huoyaoyuan** [comment]
#97830, it should be implemented in 9.0

**Comment by Liangjia0411** [comment]
thanks
